mne/viz/epochs.py

```diff
@@ -442,6 +442,194 @@ def plot_epochs(epochs, epoch_idx=None, picks=None, scalings=None,
     return fig
 
 
+def plot_epochs_concat(epochs, picks=None, scalings=None, n_epochs=8,
+                       n_channels=10, title_str='#%003i', show=True,
+                       block=False):
+    """ Visualize single trials.
+
+    Parameters
+    ----------
+
+    epochs : instance of Epochs
+        The epochs object
+    picks : array-like of int | None
+        Channels to be included. If None only good data channels are used.
+        Defaults to None
+    scalings : dict | None
+        Scale factors for the traces. If None, defaults to:
+        `dict(mag=1e-12, grad=4e-11, eeg=20e-6, eog=150e-6, ecg=5e-4, emg=1e-3,
+             ref_meg=1e-12, misc=1e-3, stim=1, resp=1, chpi=1e-4)`
+    n_epochs : int
+        The number of epochs per view.
+    n_channels : int
+        The number of channels per view.
+    title_str : None | str
+        The string formatting to use for axes titles. If None, no titles
+        will be shown. Defaults expand to ``#001, #002, ...``
+    show : bool
+        Show figure if True.
+    block : bool
+        Whether to halt program execution until the figure is closed.
+        Useful for rejecting bad trials on the fly by clicking on a
+        sub plot.
+
+    Returns
+    -------
+    fig : Instance of matplotlib.figure.Figure
+        The figure.
+    """
+    import matplotlib.pyplot as plt
+    import matplotlib as mpl
+    from matplotlib.collections import LineCollection
+    params = dict()
+    scalings = _handle_default('scalings_plot_raw', scalings)
+    scalings['eog'] = 1e6
+    color = _handle_default('color', None)
+    duration = len(epochs.times) * n_epochs
+    epoch_data = epochs.get_data()
+    if picks is None:
+        if any('ICA' in k for k in epochs.ch_names):
+            picks = pick_types(epochs.info, misc=True, ref_meg=False,
+                               exclude=[])
+        else:
+            picks = pick_types(epochs.info, meg=True, eeg=True, ref_meg=False,
+                               exclude=[])
+    if len(picks) < 1:
+        raise RuntimeError('No appropriate channels found. Please'
+                           ' check your picks')
+
+    times = epochs.times * 1e3
+    types = [channel_type(epochs.info, idx) for idx in
+             picks]
+
+    # preallocation needed for min / max scaling
+    total_channels = len(picks)
+    data = np.zeros((len(epochs.events), epochs.info['nchan'], len(times)))
+
+    for ii, epoch in enumerate(epoch_data):
+        j = 0
```

Comment:
avoid single character variable names if you can. See here: http://martinos.org/mne/stable/contributing.html